apps/logo-quiz/src/app/views/LogoVerify/LogoVerify.tsx

```diff
@@ -172,23 +172,29 @@ class LogoVerify extends React.Component<LogoVerifyProps, LogoVerifyState> {
 
   getNameButtons = (guess: QuizLetter[]): JSX.Element[] => {
     return guess.map((letter, idx) => {
+      const defaultClass = 'logo-verify__guess-btn';
+      const wrongClass = this.showGuessAsWrong() ? 'logo-verify__guess-btn--wrong' : '';
+      const correctClass = this.isVerified() ? 'logo-verify__guess-btn--correct' : '';
+      const btnClass = [defaultClass, wrongClass, correctClass].filter(Boolean).join(' ');
+
       const map: { [key: number]: () => JSX.Element } = {
         [EMPTY_SPACE.id]: () => {
-          return <span key={idx}>&nbsp;</span>;
+          return <span key={idx} className={`${btnClass} ${defaultClass}--especial`}>
+            &nbsp;
+          </span>;
         },
         [SPECIAL_CHAR.id]: () => {
-          return <span key={idx}>{letter.char}</span>;
+          return <span key={idx} className={`${btnClass} ${defaultClass}--especial`}>
+            {letter.char}
+          </span>;
         },
       };
       return map[letter.id] ? (
         map[letter.id]()
       ) : (
         <button
-          className={`logo-verify__guess-btn ${
-            this.showGuessAsWrong() ? 'logo-verify__guess-btn--wrong' : ''
-          } ${this.state.loadingGuess === idx ? 'logo-verify__guess-btn--loading' : ''} ${
-            this.isVerified() ? 'logo-verify__guess-btn--correct' : ''
-          }`}
+          className={`${btnClass}
+          ${this.state.loadingGuess === idx ? 'logo-verify__guess-btn--loading' : ''}`}
```

Comment:
I think you can also put this conditional in your cool abstraction for `btnClass`

```javascript
const loadingClass = this.state.loadingGuess === idx ? 'logo-verify__guess-btn--loading' : '';
const btnClass = [defaultClass, wrongClass, correctClass, loadingClass].filter(Boolean).join(' ');
```

Reply:
hey Carlos, I didnt add that one because I didnt want the loading animation to be shown on the special chars, what do you think?

Reply:
Sounds good